src/server/scripts/Outland/BlackTemple/boss_supremus.cpp

```diff
@@ -214,52 +188,64 @@ class boss_supremus : public CreatureScript
     }
 };
 
-class npc_volcano : public CreatureScript
+class molten_flame : public CreatureScript
 {
 public:
-    npc_volcano() : CreatureScript("npc_volcano") { }
+    molten_flame() : CreatureScript("molten_flame") { }
 
-    struct npc_volcanoAI : public ScriptedAI
+    struct molten_flameAI : public NullCreatureAI
     {
-        npc_volcanoAI(Creature* creature) : ScriptedAI(creature)
-        {
-            Initialize();
-            SetCombatMovement(false);
-        }
+        molten_flameAI(Creature* creature) : NullCreatureAI(creature) { }
 
-        void Initialize()
+        void InitializeAI() override
         {
-            wait = 3000;
+            float x, y, z;
+            me->GetNearPoint(me, x, y, z, 1, 100.0f, float(M_PI * 2 * rand_norm()));
+            me->GetMotionMaster()->MovePoint(0, x, y, z);
+            me->SetVisible(false);
```

Comment:
i dont change it, just move: https://github.com/TrinityCore/TrinityCore/blob/3.3.5/src/server/scripts/Outland/BlackTemple/boss_supremus.cpp#L80
btw, i will fix